src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Duke class, your personal assistant.
+ */
+public class Duke {
+    private TaskList taskList = new TaskList();
+    private Parser parser = new Parser();
+
+    /**
+     * Start the Duke.
+     */
+    public void start() {
+        Storage storage = new Storage("data.txt");
+        storage.fillData(taskList);
+
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        UserInterface.print("Hello from\n" + logo);
+
+        fillParser();
+
+        while (true) {
+            String input = UserInterface.read();
+            if (input.equals("bye")) {
+                try {
+                    storage.saveToStorage(taskList);
+                } catch (IOException e) {
+                    UserInterface.print(e.getMessage());
+                }
+                UserInterface.print("Bye. Hope to see you again soon!");
+                break;
+            }
+            try {
+                parser.executeCommand(input);
+            } catch (DukeException e) {
+                UserInterface.print(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Initialize Parser with built-in commands.
+     */
+    private void fillParser() {
+        parser.addCommand("todo", argument -> {
+            if (argument.equals("")) {
+                throw DukeException.INVALIDARGUMENT;
+            }
+            Task item = Task.toDo(argument);
+            taskList.addTask(item);
+            UserInterface.print("added: " + item);
```

Comment:
Maybe it would be better abstraction if all the code/methods constructing messages to print is also encapsulated into the UserInterface class instead of only using it to hold the print method. This would also be useful to make the getResponse() method for the upcoming iP tasks simpler to implement.